### ISSUE TITLE
[Bug]: Copilot returns "bad request: unknown integration"

### ISSUE THREAD

**Issue by DanielEliasib** [olimorris/codecompanion.nvim#189]
### Your `minimal.lua` config

```lua
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- set config path to enable the copilot adapter to work
vim.env["CODECOMPANION_TOKEN_PATH"] = vim.fn.expand("~/.config")

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({
    "git",
    "clone",
    "--filter=blob:none",
    "--single-branch",
    "https://github.com/folke/lazy.nvim.git",
    lazypath,
  })
end
vim.opt.runtimepath:prepend(lazypath)

-- Your CodeCompanion setup
local plugins = {
  {
    "olimorris/codecompanion.nvim",
    dependencies = {
      { "nvim-treesitter/nvim-treesitter", build = ":TSUpdate"},
      "nvim-lua/plenary.nvim",
      "hrsh7th/nvim-cmp",
      {
        "stevearc/dressing.nvim",
        opts = {},
      },
      "nvim-telescope/telescope.nvim",
    },
    opts = {
      -- Refer to: https://github.com/olimorris/codecompanion.nvim/blob/main/lua/codecompanion/config.lua
      strategies = { -- Change the adapters as required
        chat = { adapter = "copilot" },
        inline = { adapter = "copilot" },
        agent = { adapter = "copilot" },
      },
      opts = {
        log_level = "DEBUG",
      },
    },
  }
}

require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

-- Setup Tree-sitter
local ok, treesitter = pcall(require, "nvim-treesitter.configs")
if ok then
  treesitter.setup({
    ensure_installed = { "lua", "markdown", "markdown_inline", "yaml" },
    highlight = { enable = true },
  })
end

-- Setup completion
local ok, cmp = pcall(require, "cmp")
if ok then
  cmp.setup({
    mapping = cmp.mapping.preset.insert({
      ["<C-b>"] = cmp.mapping.scroll_docs(-4),
      ["<C-f>"] = cmp.mapping.scroll_docs(4),
      ["<C-Space>"] = cmp.mapping.complete(),
      ["<C-e>"] = cmp.mapping.abort(),
      ["<CR>"] = cmp.mapping.confirm({ select = true }), -- Accept currently selected item. Set `select` to `false` to only confirm explicitly selected items.
    }),
  })
end
```

### Error messages

no output 

### Log output

[DEBUG] 2024-09-09 21:11:48
Settings:
{
  env = {
    api_key = <function 1>
  },
  features = {
    text = true,
    tokens = false,
    vision = false
  },
  handlers = {
    chat_output = <function 2>,
    form_messages = <function 3>,
    form_parameters = <function 4>,
    inline_output = <function 5>,
    on_stdout = <function 6>,
    setup = <function 7>
  },
  headers = {
    Authorization = "Bearer ${api_key}",
    ["Content-Type"] = "application/json",
    ["Copilot-Integration-Id"] = "vscode-chat",
    ["editor-version"] = "Neovim/0.10.1"
  },
  name = "copilot",
  parameters = {
    max_tokens = 4096,
    model = "gpt-4o-2024-05-13",
    n = 1,
    stream = true,
    temperature = 0,
    top_p = 1
  },
  raw = { "--no-buffer", "--silent" },
  roles = {
    llm = "assistant",
    user = "user"
  },
  schema = {
    max_tokens = {
      default = 4096,
      desc = "The maximum number of tokens to generate in the chat completion. The total length of input tokens and generated tokens is limited by the model's context length.",
      mapping = "parameters",
      order = 3,
      type = "integer"
    },
    model = {
      choices = { "gpt-4o-2024-05-13" },
      default = "gpt-4o-2024-05-13",
      desc = "ID of the model to use. See the model endpoint compatibility table for details on which models work with the Chat API.",
      mapping = "parameters",
      order = 1,
      type = "enum"
    },
    n = {
      default = 1,
      desc = "How many chat completions to generate for each prompt.",
      mapping = "parameters",
      order = 5,
      type = "number"
    },
    temperature = {
      default = 0,
      desc = "What sampling temperature to use, between 0 and 2. Higher values like 0.8 will make the output more random, while lower values like 0.2 will make it more focused and deterministic. We generally recommend altering this or top_p but not both.",
      mapping = "parameters",
      order = 2,
      type = "number"
    },
    top_p = {
      default = 1,
      desc = "An alternative to sampling with temperature, called nucleus sampling, where the model considers the results of the tokens with top_p probability mass. So 0.1 means only the tokens comprising the top 10% probability mass are considered. We generally recommend altering this or temperature but not both.",
      mapping = "parameters",
      order = 4,
      type = "number"
    }
  },
  url = "https://api.githubcopilot.com/chat/completions",
  <metatable> = {
    __index = {
      extend = <function 8>,
      get_default_settings = <function 9>,
      get_env_vars = <function 10>,
      map_roles = <function 11>,
      map_schema_to_params = <function 12>,
      new = <function 13>,
      resolve = <function 14>,
      set_env_vars = <function 15>
    }
  }
}
[DEBUG] 2024-09-09 21:11:48
Messages:
{ {
    content = "You are an AI programming assistant named \"CodeCompanion\".\nYou are currently plugged in to the Neovim text editor on a user's machine.\n\nYour tasks include:\n- Answering general programming questions.\n- Explaining how the code in a Neovim buffer works.\n- Reviewing the selected code in a Neovim buffer.\n- Generating unit tests for the selected code.\n- Proposing fixes for problems in the selected code.\n- Scaffolding code for a new workspace.\n- Finding relevant code to the user's query.\n- Proposing fixes for test failures.\n- Answering questions about Neovim.\n- Running tools.\n\nYou must:\n- Follow the user's requirements carefully and to the letter.\n- Keep your answers short and impersonal, especially if the user responds with context outside of your tasks.\n- Minimize other prose.\n- Use Markdown formatting in your answers.\n- Include the programming language name at the start of the Markdown code blocks.\n- Avoid line numbers in code blocks.\n- Avoid wrapping the whole response in triple backticks.\n- Only return relevant code.\n\nWhen given a task:\n1. Think step-by-step and describe your plan for what to build in pseudocode, written out in great detail.\n2. Output the code in a single code block.\n3. You should always generate short suggestions for the next user turns that are relevant to the conversation.\n4. You can only give one reply for each conversation turn.",
    id = 1652905022,
    opts = {
      visible = false
    },
    role = "system"
  }, {
    content = "hi",
    id = 1313499796,
    opts = {
      visible = true
    },
    role = "user"
  } }
[INFO] 2024-09-09 21:11:48
Chat request started
[DEBUG] 2024-09-09 21:11:48
Copilot config dir: C:\Users\Dan\AppData\Local
[DEBUG] 2024-09-09 21:11:48
Request:
{ "-sSL", "-D", "C:\\Users\\Dan\\AppData\\Local\\Temp\\plenary_curl_b9c14a5b.headers", "-X", "POST", "-H", "Content-Type: application/json", "-H", "Editor-Version: Neovim/0.10.1", "-H", "Authorization: Bearer token", "-H", "Copilot-Integration-Id: vscode-chat", "--data-raw", "{\"model\":\"gpt-4o-2024-05-13\",\"top_p\":1,\"messages\":[{\"id\":1652905022,\"opts\":{\"visible\":false},\"content\":\"You are an AI programming assistant named \\\"CodeCompanion\\\".\\nYou are currently plugged in to the Neovim text editor on a user's machine.\\n\\nYour tasks include:\\n- Answering general programming questions.\\n- Explaining how the code in a Neovim buffer works.\\n- Reviewing the selected code in a Neovim buffer.\\n- Generating unit tests for the selected code.\\n- Proposing fixes for problems in the selected code.\\n- Scaffolding code for a new workspace.\\n- Finding relevant code to the user's query.\\n- Proposing fixes for test failures.\\n- Answering questions about Neovim.\\n- Running tools.\\n\\nYou must:\\n- Follow the user's requirements carefully and to the letter.\\n- Keep your answers short and impersonal, especially if the user responds with context outside of your tasks.\\n- Minimize other prose.\\n- Use Markdown formatting in your answers.\\n- Include the programming language name at the start of the Markdown code blocks.\\n- Avoid line numbers in code blocks.\\n- Avoid wrapping the whole response in triple backticks.\\n- Only return relevant code.\\n\\nWhen given a task:\\n1. Think step-by-step and describe your plan for what to build in pseudocode, written out in great detail.\\n2. Output the code in a single code block.\\n3. You should always generate short suggestions for the next user turns that are relevant to the conversation.\\n4. You can only give one reply for each conversation turn.\",\"role\":\"system\"},{\"id\":1313499796,\"opts\":{\"visible\":true},\"content\":\"hi\",\"role\":\"user\"}],\"n\":1,\"stream\":true,\"temperature\":0,\"max_tokens\":4096}", "--no-buffer", "--silent", "https://api.githubcopilot.com/chat/completions" }
[DEBUG] 2024-09-09 21:11:49
Copilot stdout: { "bad request: unknown integration" }
[INFO] 2024-09-09 21:11:49
Chat request completed

### Health check output

codecompanion: require("codecompanion.health").check()

- Log file: C:\Users\Dan\.repro\state\nvim-data\codecompanion.log

Plugins: ~
- OK plenary.nvim installed
- OK nvim-treesitter installed
- OK nvim-cmp installed
- OK telescope.nvim installed
- OK dressing.nvim installed

Tree-sitter parsers: ~
- OK markdown parser installed
- ERROR yaml parser not found

Libraries: ~
- OK curl installed

### Describe the bug

It seems that copilot auth with only token is not possible anymore, any request returns "bad request: unknown integration". CopilotChat.nvim seems to be doing some extra work to generate a session token from "https://api.github.com/copilot_internal/v2/token".

### Reproduce the bug

1. set copilot as you adapter
2. do any copilot request for example :CodeCompanionChat

### Final checks

- [X] I have made sure this issue exists in the latest version of the plugin
- [X] I have tested with the `minimal.lua` file from above and have shared this
- [X] I have shared the contents of the log file

**Comment by olimorris** [comment]
Thanks for raising. I've noticed this too and will investigate. Hoping it's not GitHub clamping down on none VS Code users

**Comment by olimorris** [comment]
This should now be fixed in `2.5.2` 🚀

**Comment by olimorris** [comment]
Hey @DanielEliasib just to let you know that the GitHub Secret Scanning tool alerted me to a GitHub OAuth token being leaked in this issue. Just wanted to give you the heads up
